### ISSUE TITLE
[BUG] min_chunk_size is not respected

### ISSUE THREAD

**Issue by theoden8** [chonkie-ai/chonkie#91]
**Describe the bug**
When I run semantic/SDPM chunker, min_chunk_size is not respected.

**To Reproduce**

```python
#!/usr/bin/env python3


import chonkie


text = """
Ja5uKaexug
"""
new_text = ""
for t in text.strip():
    new_text += " " + t
new_text += "\n"
text = new_text * 100


if __name__ == '__main__':
    chunker = chonkie.SDPMChunker(
        embedding_model='BAAI/bge-large-en-v1.5',
        min_chunk_size=256,
        chunk_size=512*2,
        skip_window=0,
        threshold=0.9,
    )
    chunks = [t.text for t in chunker.chunk(text)]
    print([chunker.embedding_model.count_tokens(t) for t in chunks])
    # [932, 62]
    print([len(t) for t in chunks])
    # [1953, 126]
```

**Expected behavior**

The minimum chunk size must be >= 256 tokens.

**Additional context**

Related to recently implemented #40

**Comment by theoden8** [comment]
Hey, thank you for a detailed explanation!

The hard limit for the maximum length of the chunk makes sense indeed, as anything beyond would be truncated. however, there are some reasons to want bigger chunks:

1. Parent chunks: no point having small chunks in parent chunks as they are supposed to provide wider context
2. Structured data: I use markdown delimiter first, semantic segmentation second and merger of small leftover pieces third. Having headers in chunks is more important than embedding distance outliers, and making sure bulletpoints don't break in the middle also takes precedence (third step).
3. Small chunks clutter vector database unnecessarily. Having those 3 token pieces attached to some chunk wouldn't harm.

So, basically in hybrid splitting structure and determinism take precedence for me, and semantic splitting is a nice extra that helps split the chunks not completely arbitrarily.

**Comment by bhavnicksm** [comment]
Thanks for the valuable feedback, @theoden8! 😄🫂

Totally valid, makes sense! In fact, I am working towards making a hierarchial/sequential chunking solution with benefits on Markdown, with Semantic second and then making sure that the chunks are not too small as well. It's definitely on the roadmap.

I'd have to ask you to be patient with Chonkie, since it's planned for release with version `v0.3.0` or beyond :)

Thanks 😊

(P.S. Since I can't "resolve" the issue at the moment, I will keep this open but change to a feature request instead, if you don't mind... And I will ping you here when I am done with the feature.)
